### DISCUSSION TITLE
Problem when running tests with snapshots generated on Windows OS

### DISCUSSION THREAD

**Discussion by xavimolloy** [cashapp/paparazzi#1417] [Q&A]
Hi all, it is probably something silly but we have added snapshot testing to our CI in GitHub Actions. 
tests run and are successfull locally. However when we upload the generated images to github and tests run on Github Actions tests fail with a Null pointer exception. 

I've reviewed open and closed issues and found some people had similar issues regarding the NPE and that it might be related to Git LFS. it is already implemented on the project, and I have implemented it on my device but I can't seem to find the problem code. Other colleagues in the team use Linuz and when they generate the images and upload them to the repo tests pass successfully everytime without them doing anything differently. 

The similar issues I found where only talking about how the NPE exception did not display relevant information and no information on how to resolve the issue. 

Has anyone experienced anything similar? 

thankfully the project where we have implemented the snapshot tests are public so we can share them without any worries 

This is a PR where we have created a new snapshot test that fails if generated on windows but passes when generated on Linux. 

PR  with new snapshot test:

https://github.com/dhis2/dhis2-mobile-ui/pull/236

Git hub Action that fails if image generated and pushed on windows:

https://github.com/dhis2/dhis2-mobile-ui/actions/runs/9003227033/job/24733378036

Successfull one with same proccess on linux:

https://github.com/dhis2/dhis2-mobile-ui/actions/runs/9004273183

Stacktrace: 

1m 17s
Run ./gradlew designsystem:verifyPaparazziDebug
> Task :convention-plugins:generateExternalPluginSpecBuilders UP-TO-DATE
> Task :convention-plugins:extractPrecompiledScriptPluginPlugins UP-TO-DATE
> Task :convention-plugins:compilePluginsBlocks UP-TO-DATE
> Task :convention-plugins:generatePrecompiledScriptPluginAccessors UP-TO-DATE
> Task :convention-plugins:generateScriptPluginAdapters UP-TO-DATE
> Task :convention-plugins:compileKotlin UP-TO-DATE
> Task :convention-plugins:compileJava NO-SOURCE
> Task :convention-plugins:pluginDescriptors UP-TO-DATE
> Task :convention-plugins:processResources UP-TO-DATE
> Task :convention-plugins:classes UP-TO-DATE
> Task :convention-plugins:jar UP-TO-DATE
> Task :convention-plugins:inspectClassesForKotlinIC UP-TO-DATE
Warning: Errors during XML parse:
Warning: Additionally, the fallback loader failed to parse the XML.
Checking the license for package Android SDK Platform-Tools in /usr/local/lib/android/sdk/licenses
License for package Android SDK Platform-Tools accepted.
Preparing "Install Android SDK Platform-Tools v.35.0.[1](https://github.com/dhis2/dhis2-mobile-ui/actions/runs/9003227033/job/24733378036#step:7:1)".
"Install Android SDK Platform-Tools v.35.0.1" ready.
Installing Android SDK Platform-Tools in /usr/local/lib/android/sdk/platform-tools
"Install Android SDK Platform-Tools v.35.0.1" complete.
"Install Android SDK Platform-Tools v.35.0.1" finished.
> Task :designsystem:checkKotlinGradlePluginConfigurationErrors
> Task :designsystem:preBuild UP-TO-DATE
> Task :designsystem:preDebugBuild UP-TO-DATE
> Task :designsystem:generateDebugResValues
> Task :designsystem:generateDebugResources
> Task :designsystem:packageDebugResources
> Task :designsystem:preDebugUnitTestBuild UP-TO-DATE
> Task :designsystem:javaPreCompileDebug
> Task :designsystem:processDebugManifest
> Task :designsystem:parseDebugLocalResources
> Task :designsystem:preparePaparazziDebugResources
> Task :designsystem:javaPreCompileDebugUnitTest
> Task :designsystem:generateDebugRFile
> Task :designsystem:generateDebugUnitTestStubRFile
> Task :designsystem:compileDebugKotlinAndroid
> Task :designsystem:compileDebugJavaWithJavac NO-SOURCE
> Task :designsystem:bundleLibRuntimeToJarDebug
> Task :designsystem:bundleLibCompileToJarDebug
> Task :designsystem:processDebugJavaRes
> Task :designsystem:compileDebugUnitTestKotlinAndroid
> Task :designsystem:compileDebugUnitTestJavaWithJavac NO-SOURCE
> Task :designsystem:processDebugUnitTestJavaRes

> Task :designsystem:testDebugUnitTest

org.hisp.dhis.mobile.ui.designsystem.ButtonSnapshotTest > launchButtonSnapshot FAILED
    java.lang.NullPointerException at ButtonSnapshotTest.kt:27

11 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':designsystem:testDebugUnitTest'.
> There were failing tests. See the report at: file:///home/runner/work/dhis2-mobile-ui/dhis2-mobile-ui/designsystem/build/reports/tests/testDebugUnitTest/index.html

> Task :designsystem:testDebugUnitTest FAILED

For more on this, please refer to https://docs.gradle.org/[8](https://github.com/dhis2/dhis2-mobile-ui/actions/runs/9003227033/job/24733378036#step:7:9).2.1/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
27 actionable tasks: 17 executed, 10 up-to-date

BUILD FAILED in 1m 17s
Error: Process completed with exit code 1.

In the PR you can see the line of code that is failing but it is simply the initialization of the test 

Thanks in advance!

**Comment by TWiStErRob** [comment]
Did you forget the link @xavimolloy? Also do you have the full stack, that would probably help in diagnostics.

**Reply by xavimolloy** [reply]
Hi @TWiStErRob , I most certainly did forget to post the link! apologies for that. I hadn't posted the stacktrace as the message is not very helpful similarly to this closed [ issue](https://github.com/cashapp/paparazzi/issues/1138) even after accessing the report I didn't get much more info that the NPE ate mentioned problem line.
Sorry for the late reply, based in Spain here.

I have updated the initial description with links to the PR failed actions and other details.


![image](https://github.com/cashapp/paparazzi/assets/44061143/9fd7de4a-47b2-41d8-9598-49156c693964)
